bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/handler/Enigma2Handler.java

```diff
@@ -0,0 +1,292 @@
+            updateState(channelUID, new StringType(client.getTitle()));
+        } else {
+            getLogger().info("Channel {} only accepts RefreshType. Type was {}.", channelUID, command.getClass());
+        }
+    }
+
+    private void handleDescription(ChannelUID channelUID, Command command, Enigma2Client client) {
+        if (command instanceof RefreshType) {
+            client.refreshEpg();
+            updateState(channelUID, new StringType(client.getDescription()));
+        } else {
+            getLogger().info("Channel {} only accepts RefreshType. Type was {}.", channelUID, command.getClass());
+        }
+    }
+
+    private void handlePower(ChannelUID channelUID, Command command, Enigma2Client client) {
+        if (RefreshType.REFRESH == command) {
+            client.refreshPower();
+            updateState(channelUID, client.isPower() ? OnOffType.ON : OnOffType.OFF);
+        } else if (OnOffType.ON == command) {
+            client.setPower(true);
+        } else if (OnOffType.OFF == command) {
+            client.setPower(false);
+        } else {
+            getLogger().info("Channel {} only accepts OnOffType, RefreshType. Type was {}.", channelUID, command.getClass());
+        }
+    }
+
+    public void sendRcCommand(String rcButton) {
+        getLogger().debug("sendRcCommand({})", rcButton);
+        try {
+            Enigma2RemoteKey remoteKey = Enigma2RemoteKey.valueOf(rcButton);
+            sendRcCommand(remoteKey);
+        } catch (IllegalArgumentException ex) {
+            getLogger().warn("{} is not a valid value for button - available are: {}", rcButton,
+                    Stream.of(Enigma2RemoteKey.values()).map(b -> b.name()).collect(Collectors.joining(", ")));
+        }
+    }
+
+    private void sendRcCommand(Enigma2RemoteKey remoteKey) {
+        getEnigma2Client().ifPresent(client -> client.sendRcCommand(remoteKey.getValue()));
+    }
+
+    public void sendInfo(int timeout, String text) {
+        getEnigma2Client().ifPresent(client -> client.sendInfo(timeout, text));
+    }
+
+    public void sendWarning(int timeout, String text) {
+        getEnigma2Client().ifPresent(client -> client.sendWarning(timeout, text));
+    }
+
+    public void sendError(int timeout, String text) {
+        getEnigma2Client().ifPresent(client -> client.sendError(timeout, text));
+    }
+
+    public void sendQuestion(int timeout, String text) {
+        getEnigma2Client().ifPresent(client -> client.sendQuestion(timeout, text));
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(Enigma2Actions.class);
+    }
+
+    /**
+     * Getter for Test-Injection
+     * @return Enigma2Client.
+     */
+    Optional<Enigma2Client> getEnigma2Client() {
+        return enigma2Client;
+    }
+
+    /**
+     * Getter for Test-Injection
+     * @return Enigma2Client.
+     */
+    Logger getLogger() {
+        return logger;
+    }
```

Comment:
Why are you testing logs? I would prefer if all the code used the `logger` field directly.

Reply:
Fixed